footil/tests/test_version.py

```diff
@@ -0,0 +1,146 @@
+from footil.version import GenericVersion, SemverVersion, EMPTY_VERSION
+
+from . import common
+
+
+class TestVersion(common.TestFootilCommon):
```

Comment:
These tests are a bit redundant. We have abstract version class and then two concrete version classes, which implement same methods, just use different logic.

It should be implemented somewhat similar as we have test classes for roles.

We have common test class that implement all common test cases. And if some cases must provide different results, we can override for specific version class.

This way, we have same structure/logic for tests as we have for actual version classes. And if we would have third version manager class, we could override only needed tests, without doing any copy/paste (or forgetting to test some parts).